hudi-common/src/main/java/org/apache/hudi/common/table/PartitionPathParser.java

```diff
@@ -0,0 +1,176 @@
+                                        String partitionPath,
+                                        Schema writerSchema) {
+    if (!partitionFields.isPresent()) {
+      return new Object[0];
+    }
+    return getPartitionValues(partitionFields.get(), partitionPath, writerSchema);
+  }
+
+  private static Object[] getPartitionValues(String[] partitionFields,
+                                             String partitionPath,
+                                             Schema schema) {
+    String[] parts = partitionPath.split("/");
+    int pathSegment = 0;
+    boolean hasDateField = false;
+    Object[] partitionValues = new Object[partitionFields.length];
+    for (int i = 0; i < partitionFields.length; i++) {
+      String partitionField = partitionFields[i];
+      Schema.Field field = schema.getField(partitionField);
+      // if the field is not present in the schema, we assume it is a string
+      Schema fieldSchema = field == null ? Schema.create(Schema.Type.STRING) : resolveNullableSchema(field.schema());
+      LogicalType logicalType = fieldSchema.getLogicalType();
+      if (isTimeBasedLogicalType(logicalType)) {
+        if (hasDateField) {
+          throw new IllegalArgumentException("Only one date field based partition is supported");
+        }
+        hasDateField = true;
+        int numDateDirs = parts.length - partitionFields.length + 1;
+        partitionValues[i] = inferDateValue(partitionPath, parts, pathSegment, numDateDirs, fieldSchema);
+        pathSegment += numDateDirs;
+      } else {
+        String segment = parts[pathSegment];
+        String[] segmentParts = segment.split(EQUALS_SIGN);
+        partitionValues[i] = parseValue(segmentParts[segmentParts.length - 1], fieldSchema);
+        pathSegment++;
+      }
+    }
+    return partitionValues;
+  }
+
+  @VisibleForTesting
+  static Object parseValue(String partitionValue, Schema fieldSchema) {
+    if (partitionValue.equals(DEFAULT_PARTITION_PATH) || partitionValue.equals(DEPRECATED_DEFAULT_PARTITION_PATH)) {
+      return null;
+    }
+
+    switch (fieldSchema.getType()) {
+      case STRING:
+        return PartitionPathEncodeUtils.unescapePathName(partitionValue);
+      case INT:
+        return Integer.parseInt(partitionValue);
+      case LONG:
+        return Long.parseLong(partitionValue);
+      case FLOAT:
+        return Float.parseFloat(partitionValue);
+      case DOUBLE:
+        return Double.parseDouble(partitionValue);
+      case BOOLEAN:
+        return Boolean.parseBoolean(partitionValue);
+      case BYTES:
+      case FIXED:
+        if (fieldSchema.getLogicalType() instanceof LogicalTypes.Decimal) {
+          return new java.math.BigDecimal(partitionValue);
+        } else {
+          return partitionValue.getBytes(StandardCharsets.UTF_8);
+        }
+      default:
+        throw new IllegalArgumentException("Unexpected type " + fieldSchema.getType());
+    }
+  }
+
+  private static Object inferDateValue(
```

Comment:
The date-typed partition field is not supported for bootstrap table.  So it is OK to keep the parser simple without this.

Reply:
This could be useful in other places, like in the XTable project where we need to infer the partition value

Reply:
Sg